### DISCUSSION TITLE
Low level AES_encrypt compatibility

### DISCUSSION THREAD

**Discussion by timlegge** [openssl/openssl#22042] [Q&A] [answered]
I am wondering whether the AES_encrypt function is compatible with the newer high level functions that require you to pass a cipher like EVP_EncryptInit_ex()

As in, if you have encrypted something with AES_encrypt can you get the same result or decrypt with the newer functions?

**Comment by paulidale** [comment, marked as answer]
Using the ECB cipher modes will produce the same output.

**Reply by timlegge** [reply]
Thanks, that seems to do what I want

**Reply by t8m** [reply]
Using ECB is almost never correct.

**Reply by timlegge** [reply]
lol - no but I am working on updating a perl module that used the low level api.  I want to ensure that it can continue to decrypt anything that was encrypted with the old module as I add the ability to use modern ciphers
